lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -38,6 +38,8 @@ class CodeLens < Listener
       def initialize(uri, emitter, message_queue, test_library)
         super(emitter, message_queue)
 
+        @uri = T.let(uri, String)
+        @external_listeners = T.let(external_listeners, T::Array[RubyLsp::Listener[T::Array[Interface::CodeLens]]])
```

Comment:
I think there's an extra `T::Array` in this type. Also, the listener generic parameter is the request `CodeLens` not the `Interface::CodeLens`.
```suggestion
        @external_listeners = T.let(external_listeners, T::Array[RubyLsp::Listener[CodeLens]])
```

Reply:
Nevermind, the right type is `Listener[ResponseType]` indeed. We won't be able to escape the `T.unsafe`.